sdks/corda/src/main/kotlin/com/weaver/corda/sdk/AssetTransferSDK.kt

```diff
@@ -0,0 +1,333 @@
+
+        @JvmStatic
+        fun claimPledgedAsset(
+            proxy: CordaRPCOps,
+            pledgeId: String,
+            pledgeStatusLinearId: String,
+            assetType: String,
+            assetId: String,
+            pledgerCert: String,
+            recipientCert: String,
+            getAssetAndContractIdFlowName: String,
+            createAssetStateCommand: CommandData,
+            issuer: Party,
+            observers: List<Party> = listOf<Party>()
+        ): Either<Error, SignedTransaction> {
+            return try {
+                AssetTransferSDK.logger.debug("Sending asset-claim request to Corda as part of asset-transfer.\n")
+                val signedTx = runCatching {
+
+                    val claimArgs: AssetClaimParameters = AssetClaimParameters(
+                        pledgeId, // @property pledgeId
+                        createAssetStateCommand, // @property createAssetStateCommand
+                        pledgeStatusLinearId, // @property pledgeStatusLinearId
+                        getAssetAndContractIdFlowName, // @property getAssetAndContractIdFlowName
+                        assetType, // @property assetType
+                        assetId, // @property assetIdOrQuantity
+                        pledgerCert, // @property pledgerCert
+                        recipientCert, // @property recipientCert
+                        issuer, // @property issuer
+                        observers // @property observers
+                    )
+
+                    proxy.startFlow(::AssetTransferClaim, claimArgs).returnValue.get()
+                }.fold({
+                    it.map { retSignedTx ->
+                        AssetTransferSDK.logger.debug("Claim of remote asset was successful.\n")
+                        retSignedTx
+                    }
+                }, {
+                    Left(Error("Corda Network Error: Error running AssetTransferClaim flow: ${it.message}\n"))
+                })
+                signedTx
+            } catch (e: Exception) {
+                AssetTransferSDK.logger.error("Error claiming remote asset in Corda network: ${e.message}\n")
+                Left(Error("Error claiming remote asset in Corda network: ${e.message}"))
+            }
+        }
+
+        @JvmStatic
+        fun isAssetPledgedForTransfer(
+            proxy: CordaRPCOps,
+            pledgeId: String
+        ): Boolean {
+            return try {
+                AssetTransferSDK.logger.debug("Querying if asset is pledged in Corda as part of asset-transfer.\n")
+                val isAssetPledged = proxy.startFlow(::IsAssetPledged, pledgeId)
+                    .returnValue.get()
+                isAssetPledged
+            } catch (e: Exception) {
+                AssetTransferSDK.logger.error("Error querying asset pledge state for transfer in Corda network: ${e.message}\n")
+                false
+            }
+        }
+
+        @JvmStatic
+        fun readPledgeStateByPledgeId(
```

Comment:
Any difference in terms of utility as compared to `getAssetPledgeStatus`?

Reply:
This is used in the `get-pledge-state` cli. The difference is that this function returns a `state-and-ref` value while the other returns a `state`, which is used in the cli `reclaim-pledged-asset` command.